File: app/src/main/java/com/sysu/edu/Application.java
package com.sysu.edu;

import android.content.SharedPreferences;

import androidx.preference.PreferenceManager;

import java.util.Objects;
import java.util.Set;

public class Application extends android.app.Application {
    
    @Override
    public void onCreate() {
        super.onCreate();
        SharedPreferences pm = PreferenceManager.getDefaultSharedPreferences(this);
        if ((!pm.contains("dashboard")) || Objects.requireNonNull(pm.getStringSet("dashboard", null)).isEmpty())
            pm.edit().putStringSet("dashboard", Set.of(getResources().getStringArray(R.array.values_6))).apply();
    }
}

File: app/src/main/java/com/sysu/edu/Application.kt
package com.sysu.edu

import android.app.Activity
import android.app.Application
import android.os.Bundle
import androidx.fragment.app.FragmentActivity

class Application : Application() {
	var currentActivity: FragmentActivity? = null
		private set

	override fun onCreate() {
		super.onCreate()
		registerActivityLifecycleCallbacks(object : ActivityLifecycleCallbacks {
			override fun onActivityStarted(activity: Activity) {
				currentActivity = activity as? FragmentActivity
			}

			override fun onActivityResumed(activity: Activity) {
				currentActivity = activity as? FragmentActivity
			}

			override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {}
			override fun onActivityPaused(activity: Activity) {}
			override fun onActivityStopped(activity: Activity) {}
			override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {}
			override fun onActivityDestroyed(activity: Activity) {
				if (currentActivity == activity) currentActivity = null
			}
		})
	}
}